### PR TITLE
Refactoring: PowerUps, Projectiles, Junkership and GameContainer

### PR DESCRIPTION
**Pull request by codaventure**
Made PowerUps DRY, let Projectile be polymorphic like it so desperately wanted, and removed tight coupling in the Junkership and GameContainer classes.

https://mocsarcade.github.io/starjunk/104

### CODE REVIEW

**Comment by ehgoodenough** [comment]
Thanks for making this refactor! The build isn't working for me just yet.

**Comment by codaventure** [comment]
Not sure why It didn't work the first time, but it should be all set now!

File: source/GameContainer.js [hunk 1]
```diff
@@ -31,7 +31,7 @@ export default class GameContainer extends Pixi.Container {
         this.stars = 0
         this.metrics = new Metrics(Reference.FIREBASE_URL)
         this.startedAt = Date.now()
-        this.waitingForScores = [false, false, false, false]
+        this.waitingForScores = Array(4).fill(false)
```
Comment (ehgoodenough) [inline source/GameContainer.js:34]: Huh! I've never had a use for fill before! That's neat :]

**Comment by ehgoodenough** [comment]
I'm happy with how much code was deleted; that's how you know it's a good refactor. :p

**Pull request merged**